my_game.py

```diff
@@ -45,6 +45,54 @@ def wrap(sprite: arcade.Sprite):
         sprite.center_y -= CONFIG['SCREEN_HEIGHT']
 
 
+class Shot(arcade.Sprite):
+    """
+    universal class for shot objects
+    """
+
+    def __init__(self, filename, center_x, center_y, angle, speed, range, sound=None):
+
+        super().__init__(
+            filename=filename,
+            scale=CONFIG['SPRITE_SCALING'],
+            center_x=center_x,
+            center_y=center_y,
+            angle=angle,
+            flipped_horizontally=True,
+            flipped_diagonally=True
+            )
+
+        self.speed = speed
+        self.range = range
+        self.distance_traveled = 0
+
+        self.forward(self.speed)
+
+        # play the shot sound if present
+        if sound:
+            sound.play()
+
+    def update(self):
+        """
+        move the sprite and fade
+        """
+
+        self.center_x += self.change_x
+        self.center_y += self.change_y
+
+        wrap(self)
+
+        # check if the shot traveled too far
```

Comment:
This is not a check

Reply:
well, not the line - but the comment was meant to  account for both adding to distance traveled, and checking if it exceeds range

Reply:
also... we dont need delta time - so why take unnecessary arguments?
Arcade uses update too.